common/web/lm-worker/src/main/correction/search-batcher.ts

```diff
@@ -1,63 +1,59 @@
 import { SearchNode, SearchResult } from "./distance-modeler.js";
 
-
+/**
+ * A helper class for ensuring that equally-likely corrections are returned
+ * from the correction-search on even footing.
+ *
+ * Pre-conditions:
+ * - entries should be examined in ascending-cost order.
+ */
 export class SearchBatcher {
-  currentCost = Number.MIN_SAFE_INTEGER;
-  entries: SearchResult[] = [];
+  private currentCost = Number.MIN_SAFE_INTEGER;
+  private entries: SearchResult[] = [];
 
-  // might should also include a 'base cost' parameter of sorts?
-  searchCache: {
-    [mapKey: string]: SearchNode;
-  };
-  currentReturns: {[mapKey: string]: SearchNode} = {};
+  private currentReturns: {[mapKey: string]: SearchNode} = {};
 
-  constructor(searchCache: typeof SearchBatcher.prototype.searchCache) {
-    this.searchCache = searchCache;
+  constructor() {
   }
 
-  checkAndAdd(entry: SearchNode): SearchResult[] | null {
-    var result: SearchResult[] = null;
-
-    if(entry.currentCost > this.currentCost) {
-      result = this.tryFinalize();
-
-      this.currentCost = entry.currentCost;
-    }
-
-    // Filter out any duplicated match sequences.  The same match sequence may be reached via
-    // different input sequences, after all.
-    let outputMapKey = entry.calculation.matchSequence.map(value => value.key).join('');
-
-    const searchCache = this.searchCache;
-    // First, ensure the edge has an existing 'shared' cache entry.
-    if(!searchCache[outputMapKey]) {
-      searchCache[outputMapKey] = entry;
-    }
+  /**
+   * Gets the cost of results currently batched together.
+   * All added entries should have matching `batchCost`.
+   */
+  get batchCost(): number {
```

Comment:
From the developer perspective, this splits the functionality of `checkAndAdd` into smaller pieces - that one method _was_ doing a lot.